src/commands/community/create.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  Flags,
+  loglevel,
+  orgApiVersionFlagWithDeprecations,
+  requiredOrgFlagWithDeprecations,
+  SfCommand,
+} from '@salesforce/sf-plugins-core';
+import { CommunityNameValueParser } from '../../shared/community/commands/CommunityNameValueParser';
+import { ConnectExecutor } from '../../shared/connect/services/ConnectExecutor';
+import { CommunityCreateResource } from '../../shared/community/connect/CommunityCreateResource';
+import { CommunityCreateResponse } from '../../shared/community/defs/CommunityCreateResponse';
+import { applyApiVersionToOrg } from '../../shared/utils';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/plugin-community', 'create');
+
+const MESSAGE_KEY = 'message';
+const NAME_KEY = 'name';
+const ACTION_KEY = 'action';
+
+/**
+ * A command to create a community.
+ * This is just an sfdx wrapper around the community create connect endpoint
+ */
+export class CommunityCreateCommand extends SfCommand<CommunityCreateResponse> {
+  public static readonly aliases = ['force:community:create'];
+  public static readonly summary = messages.getMessage('summary');
+  public static readonly description = messages.getMessage('description');
+  public static readonly examples = messages.getMessages('examples');
+  public static readonly flags = {
+    name: Flags.string({
+      char: 'n',
+      summary: messages.getMessage('flags.name.description'),
+      description: messages.getMessage('flags.name.longDescription'),
+      required: true,
+    }),
+    'template-name': Flags.string({
+      char: 't',
+      summary: messages.getMessage('flags.templateName.description'),
+      description: messages.getMessage('flags.templateName.longDescription'),
+      required: true,
+      aliases: ['templatename'],
+    }),
+    'url-path-prefix': Flags.string({
+      char: 'p',
+      summary: messages.getMessage('flags.urlPathPrefix.description'),
+      description: messages.getMessage('flags.urlPathPrefix.longDescription'),
+      required: true,
+      aliases: ['urlpathprefix'],
+    }),
+    description: Flags.string({
+      char: 'd',
+      summary: messages.getMessage('flags.description.description'),
+      description: messages.getMessage('flags.description.longDescription'),
+    }),
+    'target-org': requiredOrgFlagWithDeprecations,
+    loglevel,
+    'api-version': orgApiVersionFlagWithDeprecations,
+  };
+
+  public static readonly validationPatterns: string[] = [
+    // Exact matches
+    'name',
+    'urlPathPrefix',
+    'templateName',
+    'description',
+
+    // templateParams.*, but must be only word characters (e.g. no spaces, special chars)
+    'templateParams(\\.\\w+)+',
+  ];
+  private logger: Logger;
+
+  public async run(): Promise<CommunityCreateResponse> {
+    this.logger = Logger.childFromRoot(this.constructor.name);
+    const { flags, argv } = await this.parse(CommunityCreateCommand);
```

Comment:
Does this require setting `public static readonly varargs=true` on the command? I'm not sure how it works anymore and see it as removed from the original

Reply:
Good catch! oclif/core added command property `strict: boolean` which is set to true by default. This suppresses args/varargs during parse. I need to set it to false for this command.